core/src/main/java/io/grpc/internal/AbstractServerStream.java

```diff
@@ -149,8 +154,11 @@ protected void inboundHeadersReceived(Metadata headers) {
       }
     }
     if (headers.containsKey(GrpcUtil.MESSAGE_ACCEPT_ENCODING_KEY)) {
-      pickCompressor(
+      Compressor c = pickCompressor(
           ACCEPT_ENCODING_SPLITER.split(headers.get(GrpcUtil.MESSAGE_ACCEPT_ENCODING_KEY)));
+      if (c != null) {
```

Comment:
If messageEncoding was set earlier, and the server receive a header advertises encodings none of which the server supports, shouldn't the server reset messageEncoding?